src/platform/plugins/shared/console/server/lib/spec_definitions/js/retriever.ts

```diff
@@ -105,6 +105,28 @@ export const retriever = (specService: SpecDefinitionsService) => {
       match_criteria: {},
       rank_window_size: 10,
     },
+    pinned: {
+      __template: {
+        retriever: {},
+        ids: [],
+        docs: [],
+        match_criteria: {},
+      },
+      retriever: {
+        __scope_link: '.',
+      },
+      // Only one of 'ids' or 'docs' should be used at a time
+      ids: {
+        __description:
+          "Specify an array of document IDs to pin. Only one of 'ids' or 'docs' can be used.",
+      },
+      docs: {
```

Comment:
the same for `docs` - we should autocomplete with an array.

I think it's easier to just put `ids: [ ], docs: [ ]`